.pre-commit-config.yaml

```diff
@@ -87,10 +87,19 @@ repos:
     #     stages: [commit, push]
     -   id: trailing-whitespace         # Trims trailing whitespace.
         stages: [commit, push]
-
+# First pass: check format
 -   repo: https://github.com/pre-commit/mirrors-yapf
-    rev: v0.22.0
+    rev: v0.25.0
     hooks:
     -   id: yapf
+        name: Check format by yapf
         args: ['-dpr']
         stages: [commit, push]
+# Second pass: format in-place
```

Comment:
hmm why not just do the in-place format and be done with it?

Reply:
Oh I see `-dpr` returns non-zero if it finds a change, which will fail the CI. `-ipr` will not return non-zero on a change. never mind.